luigi/task.py

```diff
@@ -911,14 +911,19 @@ def flatten(struct):
 def flatten_output(task):
     """
     Lists all output targets by recursively walking output-less (wrapper) tasks.
-
-    FIXME order consistently.
     """
-    r = flatten(task.output())
-    if not r:
-        for dep in flatten(task.requires()):
-            r += flatten_output(dep)
-    return r
+
+    output_tasks = []
+    tasks_to_process = deque([task])
+    while tasks_to_process:
+        current_task = tasks_to_process.popleft()
+        if flatten(current_task.output()):
+            if current_task not in output_tasks:
```

Comment:
From my point of view, a list is efficient enough as I've never seen a use-case of more than 30-40 tasks required by one wrapper task (it's working fast even for hundreds). Unfortunately, python doesn't have a built-in ordered set, and I wanted to preserve some explainable order. 

To provide order and speed similar to set, I can use a dict (OrderedDict because of support for python 3.6) hack:
```
output_tasks = OrderedDict()
...
output_tasks[current_task] = None
```

A set will be the best choice if an order is not required.